src/mlpack/methods/ann/layer/convolution_impl.hpp

```diff
@@ -253,41 +239,47 @@ void Convolution<
     InputDataType,
     OutputDataType
 >::Backward(
-    const arma::Mat<eT>& /* input */, const arma::Mat<eT>& gy, arma::Mat<eT>& g)
+    const arma::Mat<eT>& input, const arma::Mat<eT>& gy, arma::Mat<eT>& g)
 {
   arma::cube mappedError(((arma::Mat<eT>&) gy).memptr(), outputWidth,
       outputHeight, outSize * batchSize, false, false);
+  arma::cube inputTemp(((arma::Mat<eT>&) input).memptr(), inputWidth,
+      inputHeight, inSize * batchSize, false, false);
 
-  g.set_size(inputWidth * inputHeight * inSize, batchSize);
-  gTemp = arma::Cube<eT>(g.memptr(), inputWidth, inputHeight,
-      inSize * batchSize, false, false);
+  g.set_size(inputTemp.n_rows * inputTemp.n_cols * inSize, batchSize);
+  gTemp = arma::Cube<eT>(g.memptr(), inputTemp.n_rows, inputTemp.n_cols,
+                         inputTemp.n_slices, false, false);
   gTemp.zeros();
-
-  for (size_t outMap = 0, outMapIdx = 0, batchCount = 0; outMap <
-      outSize * batchSize; outMap++)
+  #pragma omp parallel
```

Comment:
I didn't mean this, I mistakenly put it there thinking the parallel mark did something else. However, since you pointed it out, I realize I made an error. Will fix it